java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java

```diff
@@ -107,6 +112,23 @@ public static void printResults(
     public static void testClientSetGet(
             Supplier<Client> clientCreator, BenchmarkingApp.RunConfiguration config, boolean async) {
         for (int concurrentNum : config.concurrentTasks) {
+            // same as Executors.newCachedThreadPool() with a RejectedExecutionHandler for robustness
+            ExecutorService executor =
+                    new ThreadPoolExecutor(
+                            0,
+                            Integer.MAX_VALUE,
+                            60L,
+                            TimeUnit.SECONDS,
+                            new SynchronousQueue<Runnable>(),
```

Comment:
I'm trying to understand why a synchronous queue is the right queue. It blocks insertions until there are removals. This means that some working threads will be blocked merely on trying to add new tasks to the list. Isn't this just backpressure to prevent more tasks being created?

Reply:
The synchronous queue more closely resembles what we are doing in Python/Node.  Its better for TPS, but less-ideal for latency.  I think https://www.baeldung.com/java-executors-cached-fixed-threadpool does a good job explaining the differences. 

The LinkedBlockingQueue will allocate tasks to threads.  In our case, when there are tasks >> threads, the remainder get put into a blocking queue and do not execute any work. This means limited swapping, which can lead to more waiting on I/O operations. This reduces latency on tasks, but can limit TPS. 

The SynchronousQueue will swap out tasks that are suspended wherever possible, which means there is more CPU time allocated to a more even distribution of tasks.  This increases TPS by parallelizing as much as possible, but could mean increasing latency.  

The latter is what Python/Note concurrency seems to be doing (based on our observations).

Reply:
https://stackoverflow.com/questions/78145817/yielding-control-in-java-asynchronous-code

I try crowdsource a solution.